libp2p/protocols/pubsub/pubsub.nim

```diff
@@ -565,7 +585,13 @@ method publish*(
   if p.triggerSelf:
     await handleData(p, topic, data)
 
-  return 0
+  return ok(0)
+
+proc publish*(
+    p: PubSub, topic: string, data: seq[byte]
+): Future[int] {.async: (raises: []), public.} =
+  return (await p.doPublish(topic, data)).valueOr:
```

Comment:
the new structure of the code introduces several additional copies of the data - this is not acceptable for a frequently-called functions like `publish`, specially because `data` may be large.

Every time `seq` is used with `async`, a copy is made - every time `Result`+`valueOr` is used, a copy is made, etc.

In _most cases_ this doesn't matter, but libp2p is a core library where small changes like this have an overall negative effect on downstream consumers and care must be taken.